test/e2e/framework.go

```diff
@@ -1392,7 +1394,46 @@ func (data *TestData) deleteClickHouseOperator(chOperatorYML string) error {
 	return nil
 }
 
+func (data *TestData) killProcesses(namespace, podName, containerName, processName string) error {
+	cmds := []string{"pgrep", "-f", processName}
+	stdout, stderr, err := data.RunCommandFromPod(namespace, podName, containerName, cmds)
+	if err != nil {
+		if strings.Contains(err.Error(), "does not have a host assigned") {
+			return nil
+		}
+		return fmt.Errorf("error when getting pid of '%s', stderr: <%v>, err: <%v>", processName, stderr, err)
+	}
+	cmds = []string{"kill", "-SIGINT", strings.TrimSpace(stdout)}
+	_, stderr, err = data.RunCommandFromPod(namespace, podName, containerName, cmds)
+	if err != nil {
+		return fmt.Errorf("error when sending SIGINT signal to '%s', stderr: <%v>, err: <%v>", processName, stderr, err)
+	}
+	return nil
+}
+
+func (data *TestData) killProcessesOnPods() error {
+	listOptions := metav1.ListOptions{}
+
+	pods, err := data.clientset.CoreV1().Pods(flowVisibilityNamespace).List(context.TODO(), listOptions)
+	if err != nil {
+		return fmt.Errorf("failed to list flow-visibility pods: %v", err)
+	}
+	for _, pod := range pods.Items {
+		podName := pod.Name
+		if strings.Contains(podName, "chi-clickhouse-clickhouse") {
+			err = data.killProcesses("flow-visibility", podName, clickHouseMonitorContName, clickHouseMonitorContName)
+		} else if strings.Contains(podName, theiaManagerContName) {
+			err = data.killProcesses("flow-visibility", podName, theiaManagerContName, theiaManagerContName)
+		}
```

Comment:
Why do we only kill processes for Theia manager and ClickHouse monitor?

Reply:
Their binaries have been compiled with -cover flag which allows a coverage report to be generated when the program has exited or finished running. Since both processes don't stop until instructed to, I added the instruction before teardown so that coverage files can be collected before pods and nodes are torn down and files are lost.